src/include/cmark-gfm.h

```diff
@@ -807,7 +807,7 @@ const char *cmark_version_string(void);
  * John MacFarlane, Vicent Marti,  Kārlis Gaņģis, Nick Wellnhofer.
  */
 
-#ifndef CMARK_NO_SHORT_NAMES
+#if !defined(CMARK_NO_SHORT_NAMES) && !defined(_WIN32)
```

Comment:
Could you explain why this is needed?

Reply:
> However, due to name collisions between cmark's "short" symbol names and some symbols in the Win32 API, short names are disabled universally on Windows.

`msxml.h` defines an enum `tagDOMNodeType` with several `NODE_{kind}` symbols of its own, a couple of which (specifically `NODE_TEXT` and `NODE_DOCUMENT`) collide with these short symbol names. While developing, i decided to drop these short symbols in this situation, since i was previously unaware of the "short names" feature.

Looking at this again today, it seems like this header is dropped under `WIN32_LEAN_AND_MEAN`, but the required SRWLock APIs are kept - should i try to do this instead? I don't know what specific `NO{feature}` to apply to drop MSXML specifically.

Reply:
Yes, `WIN32_LEAN_AND_MEAN` is generally encouraged, so you should definitely do that.  The fact that it happens to also fixes the issue, is a nice bonus :).